third_party/xkcp/add_main_license.patch

```diff
@@ -0,0 +1,80 @@
+diff --git a/LICENSE b/LICENSE
+new file mode 100644
+index 0000000..662709a
+--- /dev/null
++++ b/LICENSE
+@@ -0,0 +1,74 @@
++Note: This repo uses only some part of the code from XKCP library, therefore only the corresponding license notices are given below for the used part of the source code. The full version of the library and its comprehensive license can be found at https://github.com/XKCP/XKCP.
++
++The redistribution and use of this software (with or without changes) is allowed without the payment of fees or royalties provided that the terms of the licenses of the different source files used is respected. Most of the source and header files in the XKCP are released to the public domain and associated to the CC0 (http://creativecommons.org/publicdomain/zero/1.0/) deed, but there are exceptions.
```

Comment:
Hmm is this something you wrote? Do we need this at all?

Can't you just have a patch file that deletes all components that you do not use? Then you will only have vendored in what you need, and their respective LICENSE files.

Reply:
That's also what I initially did but it has the following problem:

The main LICENSE file also has a GPL header for few files in the XKCP repo (we do not need these files so I excluded them in bazel config). When I make a patch on top of the original LICENSE file, the patch file then brings the GPL header into the patch file itself, since the diff file contains the lines removed from the original file. In short, since we do not bring any GPL-licensed file into the repo, I think it is best to avoid GPL headers as well (even if they are patches that *remove* those headers/files).

Reply:
[.](https://github.com/lowRISC/opentitan/pull/16906#issuecomment-1384027314)